cpp/ql/src/Security/CWE/CWE-311/CleartextStorage.c

```diff
@@ -1,12 +1,31 @@
-void writeCredentials() {
-  char *password = "cleartext password";
-  FILE* file = fopen("credentials.txt", "w");
-  
+#include <sodium.h>
+#include <stdio.h>
+#include <string.h>
+
+void writeCredentialsBad(FILE *file, const char *cleartextCredentials) {
   // BAD: write password to disk in cleartext
-  fputs(password, file);
-  
-  // GOOD: encrypt password first
-  char *encrypted = encrypt(password);
-  fputs(encrypted, file);
+  fputs(cleartextCredentials, file);
```

Comment:
In the bad example we use `fputs`, but in the good example we now use `fwrite`. Is that intentional?

Reply:
Yes, it's because `fputs` writes a null terminated string, but after encryption we have a block of data and a length, not a null terminated string - so `fwrite` is appropriate.